### DISCUSSION TITLE
Using @Default annotation and required together ....

### DISCUSSION THREAD

**Discussion by stephenhauck** [rrousselGit/freezed#1032] [Q&A]
When I try to make a field / property have a default with the @Default annotation it does not allow me to also mark it as required ... 
Just confirming this is because by having a default it will always have a value ?

**Comment by SunlightBro** [comment]
This is intended, take this NON-freezed as an example:
```dart
class DataClass {
  final String name;
  final int id;
  final double value;

  DataClass({
    required this.name,
    required this.id,
    this.value = 100.0, // you can't add a `required` on value
  });
}
```

[default_value_on_required_parameter](https://dart.dev/tools/diagnostic-messages?utm_content=default_value_on_required_parameter#default_value_on_required_parameter)

**Reply by SunlightBro** [reply]
If I remember correctly this `@Default()` annotation is used, so that all the generated constructors and functions (also the one's from json_serilizable), will know about this default value.
